app.py

```diff
@@ -10,13 +10,15 @@
 import config
 
 app = Flask(__name__)
+app.secret_key = config.secret_key
 
 @app.route('/')
 def location():
 
     data = get_ip_info()
```

Comment:
Can add a check here if the session data exists from a previous request, if so use that instead of making the API calls again.